src/main/java/taxi/dao/DriverDaoImpl.java

```diff
@@ -101,14 +103,37 @@ public boolean delete(Long id) {
         }
     }
 
+    public Optional<Driver> findByLogin(String login) {
+        String query = "SELECT id, name, license_number,login,password\n"
+                + "FROM drivers\n"
+                + "where login = ?;";
+
+        Driver driver = new Driver();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setString(1, login);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                driver = parseDriverFromResultSet(resultSet);
+            }
+            return Optional.of(driver);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't find driver by login " + login, e);
+        }
```

Comment:
```suggestion
        String query = "SELECT id, name, license_number,login,password\n"
                + "FROM drivers\n"
                + "where login = ?;";

        try (Connection connection = ConnectionUtil.getConnection();
                PreparedStatement statement = connection.prepareStatement(query)) {
            statement.setString(1, login);
            ResultSet resultSet = statement.executeQuery();
            if (resultSet.next()) {
                return Optional.of(parseDriverFromResultSet(resultSet));
            }
        } catch (SQLException e) {
            throw new DataProcessingException("Couldn't find driver by login " + login, e);
        }
        return Optional.empty();
```

you are returning empty user if the login wasn't present in db